protocol/signalfx/log_hecv1_json.go

```diff
@@ -0,0 +1,66 @@
+package signalfx
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mailru/easyjson"
+	"github.com/signalfx/golib/v3/log"
+	"github.com/signalfx/golib/v3/logsink"
+	signalfxformat "github.com/signalfx/ingest-protocols/protocol/signalfx/format"
+)
+
+const (
+	// resources
+	host   = "host.hostname"
+	source = "service.name"
+	// attributes
+	sourceType = "source.type"
+	index      = "index"
+)
+
+// JSONDecoderHECV1 implements sink decoder for hecv1 splunk log events
+type JSONDecoderHECV1 struct {
+	Sink   logsink.Sink
+	Logger log.Logger
+}
+
+// Read implements interface method for decoding incoming json hecv1 format and convert it into golib/logsink.Log format
+func (decoder *JSONDecoderHECV1) Read(ctx context.Context, req *http.Request) (err error) {
+	var logHECV1List signalfxformat.JSONLogHECV1List
+	if err = easyjson.UnmarshalFromReader(req.Body, &logHECV1List); err == nil {
+
+		logs := make(logsink.Logs, len(logHECV1List))
+		for idx, lEntry := range logHECV1List {
+			e := &logsink.Log{
+				TimeStamp: uint64(*lEntry.Time),
```

Comment:
1. Can `lEntry.Time` be `nil` if missing?
2. I am not sure this is correct. Timestamp in HEC is fractional seconds (BTW, it seems the type of `JSONLogHECV1.Time` is incorrectly declared as `int64` while it probably needs to be a `string`). Timestamp in `Log` supposedly is nanoseconds (I am assuming you decided to adopt OpenTelemetry-like timestampts). So you probably need to translate fractional seconds to nanoseconds.

Reply:
JSONLogHECV1.Time will be fixed in subsequent commit.

Reply:
yes `lEntry.Time` can be nil based on the model and hence added that check